### PR TITLE
Plot environments in order of the modification date of their directory

### PR DESCRIPTION
**Pull request by mbertheau**
This changes plot_train.py to plot environments in order of the time of their
most recent content modification. This way, newer models appear on top.

## Description

Before, the environments were plotted in random order as returned by
listdir().

Now they are sorted by the time of most recent content modification.

This way, newer enviroments appear on top.

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

I found it more useful to have the models be plotted in a deterministic order.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

**Comment by qgallouedec** [comment]
I'd say that's a good idea (next time, prefer to open an issue to suggest an enhancement).

How about sorting by name instead of by last modification? That would give the user more flexibility to choose the order him/herself, and by default the curves would be sorted by order of modification as well (since the name is incremented with each new record).

**Comment by mbertheau** [comment]
> I'd say that's a good idea (next time, prefer to open an issue to suggest an enhancement).

If you don't mind me asking: I see the benefit of opening an issue first when you want to get buy-in for an idea or a feature before you put in time and effort to implement it. In this case, however, the changes are very small and low-effort, and are already done. In that case, why open an issue first?

> How about sorting by name instead of by last modification? That would give the user more flexibility to choose the order him/herself, and by default the curves would be sorted by order of modification as well (since the name is incremented with each new record).

I had that at first, but I was at experiment numbers > 9 already, so sorting my name put MyEnv-v1-10 before MyEnv-v1-2 and so on. That's why I settled on the modification date.

You're correct, however, that sorting by name makes it easier for somebody to influence the order themselves.

I could change the code to emit MyEnv-v1-001 instead of -1, that would address that concern. WDYT?

**Comment by araffin** [comment]
> How about sorting by name instead of by last modification? 
> That's why I settled on the modification date.

yes, that was what I expected, I think sorting by modification date is a good solution in that case.
(we could also manually sort but this is a bit more hacky)

**Comment by qgallouedec** [comment]
> (we could also manually sort but this is a bit more hacky)

For this specific need, it requires to locally modify `plot_train.py`, right?

File: rl_zoo3/plots/plot_train.py [hunk 1]
```diff
@@ -58,11 +58,9 @@ def plot_train():
...
         dirs.extend(
-            [
-                os.path.join(log_path, folder)
-                for folder in os.listdir(log_path)
-                if (env in folder and os.path.isdir(os.path.join(log_path, folder)))
-            ]
+            entry.path
+            for entry in sorted(os.scandir(log_path), key=lambda e: e.stat().st_mtime)
+            if env in entry.name and entry.is_dir()
         )
```
Comment (qgallouedec) [inline rl_zoo3/plots/plot_train.py:64]: Make it a little easier to understand.

```suggestion
        # Sort by last modification
        entries = sorted(os.scandir(log_path), key=lambda entry: entry.stat().st_mtime)
        dirs.extend(entry.path for entry in entries if env in entry.name and entry.is_dir())
```

**Comment by araffin** [comment]
@qgallouedec are you ok with the changes or is something missing?

> For this specific need, it requires to locally modify plot_train.py, right?

which specific need? sorting the folders by their numbers?
yes for now.

**Review by qgallouedec — APPROVED** [review]

**Pull request merged**